klient/k8s/watcher/watch.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watcher
+
+import (
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// EventHandlerFuncs is an adaptor to let you easily specify as many or
+// as few of functions to invoke while getting notification from watcher
+type EventHandlerFuncs struct {
+	addFunc    func(obj interface{})
+	updateFunc func(newObj interface{})
+	deleteFunc func(obj interface{})
+	watcher    watch.Interface
+}
+
+// EventHandler can handle notifications for events that happen to a resource.
+//  * OnAdd is called when an object is added.
+//  * OnUpdate is called when an object is modified.
+//  * OnDelete will get the final state of the item if it is known.
+// Start will be waiting for the events notification which is responsible
+// for invoking the registered user defined functions
+type EventHandler interface {
+	OnAdd(obj interface{})
```

Comment:
These methods, `OnAdd`, `OnDelete`, and `OnUpdate` is redundant with RegisterAdd, RegisterDelete, and RegisterUpdate methods.  There should only be one set. When called, they should just save the function to be called during event processing.